data_model/src/permissions.rs

```diff
@@ -75,9 +79,32 @@ impl Registered for PermissionTokenDefinition {
 }
 
 impl PermissionTokenDefinition {
-    /// Construct new `PermissionTokenDefinition`
-    pub fn new(id: Id) -> Self {
-        Self { id }
+    /// Construct new [`PermissionTokenDefinition`]
+    #[inline]
+    pub fn new(id: <PermissionTokenDefinition as Identifiable>::Id) -> Self {
+        Self {
+            id,
+            params: btree_map::BTreeMap::default(),
+        }
+    }
+
+    /// Add parameters to the `PermissionToken` replacing any previously defined
+    #[inline]
+    #[must_use]
+    pub fn with_params(
+        mut self,
+        params: impl IntoIterator<Item = (Name, crate::ValueKind)>,
+    ) -> Self {
+        self.params = params.into_iter().collect();
+        self
+    }
+
+    /// Get an iterator over parameters of the `PermissionToken`
+    ///
+    /// Values returned from the iterator are guaranteed to be in alphabetical order.
+    #[inline]
+    pub fn params(&self) -> impl ExactSizeIterator<Item = (&Name, &crate::ValueKind)> {
+        self.params.iter()
```

Comment:
this function is not the same as the one on lines 178-185 but the doc comment is the same. Shouldn't you mention value kind here?

Reply:
Indeed, thanks